### DISCUSSION TITLE
CURRENT WORKING VERSION

### DISCUSSION THREAD

**Discussion by TheHugeFluff** [victornpb/undiscord#534] [1. Questions & Answers]
THE CURRENT WORKING VERSION OF THIS IS
https://greasyfork.org/en/scripts/406540-undiscord?version=1159585

I tested and I had to update the link 5.2.0 works for me flawlessly

**Comment by sramem2at3a** [comment]
Thank you, it's actually working, but returning api error after deleting about 25 messages.

**Reply by TheHugeFluff** [reply]
If you do it over again it will work. I'm pretty sure it's the API limit or what ever it's called that prevents spam

**Reply by potato899** [reply]
Doesn't work for me, the icon of the trash bin to open the script settings doesn't appear since forever, not in this version nor in the updated one.

Edit: nevermind, the damn script auto-updated everytime so that's why it didn't work.

**Comment by sramem2at3a** [comment]
@TheHugeFluff Yeah doing it each and every time it failes did the job although it's still lengthy and breaks the purpose, but I'll take 25 at once rather than deleting manually.
Just writing for someone who might face the same as @potato899 has. The trash bin icon didn't show at first, i turned off all ad block and popup blocker scripts and it appeared.

**Comment by varlogtim** [comment]
@sramem2at3a and @TheHugeFluff , I was able to get around the annoying re-clicking of start by waiting several second and then calling `recurse()` again instead of ended.

Also, I introduced a random delay which seems to help things along.
```

            log.verb(`Searching next messages in ${searchDelay}ms...`, (offset ? `(offset: ${offset})` : ''));
            await wait(searchDelay + Math.floor(Math.random() * 600));

            if (stopHndl && stopHndl() === false) return end(log.error('Stopped by you!'));

            return await recurse();
        } else {
            if (total - offset > 0) log.warn('Ended because API returned an empty page. I am sleeping. 10 seconds.');
            await wait(10000);
            if (stopHndl && stopHndl() === false) return end(log.error('Stopped by you!'));
            return await recurse();
            //return end();
        }
    }
```

**Reply by nihiluis** [reply]
Where did you add this? There's no recurse function in the working version mentioned by op.

**Comment by etozheDT** [comment]
![image](https://github.com/victornpb/undiscord/assets/143742275/61fe3262-f925-4da6-a275-7885228aede3)
How to fix this issue then starting script?

**Reply by TheHugeFluff** [reply]
well I fixed that by clicking the author ID
![Screen Shot 2023-08-31 at 1 29 42 PM](https://github.com/victornpb/undiscord/assets/73508130/89edfcec-fa17-46cd-9726-ce09a6b263f3)
![Screen Shot 2023-08-31 at 1 31 30 PM](https://github.com/victornpb/undiscord/assets/73508130/9fd1e5d0-88f0-4a2e-b206-bbc9738b95d3)

**Comment by RUSKuks** [comment]
Greetings. So far with problems but a working version.
https://pastebin.com/raw/PsnJhACM
https://youtu.be/dEx_5LyY3iI

**Comment by Astral-Crescent** [comment]
Literally the script no longer works at all and the author needs to stop pretending that it is. 

No matter what I do it now either says 0 messages or it tries to delete every message since the dawn of time. I have used this script for YEARS without issue until this last month, lmao.

**Comment by PetaPat** [comment]
<img width="1433" alt="Screenshot 2023-12-17 at 3 45 46 AM" src="https://github.com/victornpb/undiscord/assets/154003247/8aa59c99-e3df-4f6e-89e2-2884e2408fae">
what does this mean what do i do nothing is deleting

**Reply by RUSKuks** [reply]
You forgot to specify Channel ID

**Reply by PetaPat** [reply]
Even if i do it it doesnt seem to wanna work 😢
<img width="1393" alt="Screenshot 2023-12-28 at 3 23 18 AM" src="https://github.com/victornpb/undiscord/assets/154003247/244cb58d-1139-4fca-88fa-01c17d4c0250">

**Reply by PetaPat** [reply]
im using version 5.20 but it still says this meesage
